Source/ASDisplayNode.mm

```diff
@@ -590,6 +595,10 @@ - (void)_locked_loadViewOrLayer
   } else {
     TIME_SCOPED(_debugTimeToCreateView);
     _view = [self _locked_viewToLoad];
+    if ([self supernode] == nil) {
+      // Move to supernode wil propagateDown other way
```

Comment:
I assume this comment tries to explain why we need the super node check above, but it isn't clear to me. Can you elaborate?

Reply:
If I understand correctly, what you want to check here is whether this node is the root node within a view controller. If that's the case, you can simply check `node.isViewControllerRoot` is what you want. This flag is set automatically by ASViewController. If you don't use ASViewController, you need to set it yourselves.

Reply:
In some way, I use UIViewController and add subnodes later, these nodes will face the same problem, thus `_flags.isViewControllerRoot` might not be the best choice.

change the ivar directly because I wanted less logic during `_locked_loadViewOrLayer` which might have less side effect for new code. And I debug it find out here is quite early like init, and this value updated will propagate down later.
  
 `setPrimitiveTraitCollection` also works and it's little bit heavy I think.